src/App.tsx

```diff
@@ -17,18 +61,39 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                options={filterOptions}
+                filter={filter}
+                onSelect={selectHandler}
+                searchValue={searchValue}
+                onChange={searchHandler}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todosFromServer.length > 0
+                ? (
+                  <TodoList
+                    todos={visibleTodos}
+                    selectedTodo={selectedTodo}
+                    setSelectedTodo={setSelectedTodo}
+
+                  />
+                )
+                : (
+                  <Loader />
+                )}
```

Comment:
show Loader only when todos is fetching. If there is no todos there should be empty state text